cs/battle.go

```diff
@@ -972,6 +973,21 @@ func (b *battle) regenerateShields(token *battleToken) {
 	}
 }
 
+func updateBestPositions(better bool, new BattleVector, bestMoves []BattleVector) []BattleVector {
+	// center of battle board is at (4.5, 4.5)
+	newsquared := (2*new.X - 9)*(2*new.X - 9) + (2*new.Y - 9)*(2*new.Y - 9)
+	oldsquared := (2*bestMoves[0].X - 9)*(2*bestMoves[0].X - 9) + (2*bestMoves[0].Y - 9)*(2*bestMoves[0].Y - 9)
+
+	if better || newsquared < oldsquared {
+		// this move is strictly better than other moves based on damage or proximity to center
+		bestMoves = nil
+	} else if oldsquared < newsquared {
+		// damage is the same but proximity is worse
+		return bestMoves
+	}
+	return append(bestMoves, new)
+}
```

Comment:
Can you write a unit test for this function? If you are using vscode, there is a `Go: Generate Unit Tests for function` that automatically scaffolds out a table test with args, and wanted return value. It makes it easier to reason about what the function is doing if I can see a set of inputs and their corresponding outputs.